Sources/SwiftSyntax/Raw/RawSyntaxTokenView.swift

```diff
@@ -65,6 +65,29 @@ public struct RawSyntaxTokenView {
     }
   }
 
+  @_spi(RawSyntax)
+  public func wholeText<T>(_ body: (SyntaxText) -> T) -> T {
```

Comment:
Could we instead have a function to get the `UnicodeScalar` for a particular offset?

Reply:
The current diagnostics only reference a single character but I could imagine that future diagnostics might reference an entire substring, e.g. from the offset to the end of the token, so I think the added flexibility of this API is good.

Reply:
That's true... in that case Doug added a `withEachSyntaxText` that you could use instead. It's a ... little more involved, but would avoid the temporary buffer creation (well, we still make it for leading/trailing).

Alternatively given this is only ever used in the error case, you could just use `syntaxTextBytes` (which uses `withEachSyntaxText` but writes it into an array).

This function is just sort of half way, which I don't really see the point of. Either we care about performance, in which case we should use `withEachSyntaxText`. Or we don't and thus `syntaxTextBytes` is simpler.

Reply:
Using `syntaxTextBytes` sounds good to me.